draft-ietf-quic-recovery.md

```diff
@@ -301,63 +302,117 @@ all subsequent ACK frames containing them could be lost. In this case, the
+lifetime of the connection (min_rtt), an exponentially-weighted moving average
+(smoothed_rtt), and the variance in the observed RTT samples (rttvar).
+
+## Generating RTT samples {#latest-rtt}
+
+An endpoint generates an RTT sample on receiving an ACK frame that meets the
+following two conditions:
+
+- the largest acknowledged packet number is higher than any previously seen, and
+
+- at least one of the newly acknowledged packets was ack-eliciting.
+
+The RTT sample, latest_rtt, is generated as the time elapsed since the largest
+acknowledged packet was sent:
+
 ~~~
 latest_rtt = ack_time - send_time_of_largest_acked
 ~~~
 
-First RTT sample:
+An endpoint uses only locally observed times in generating RTT samples and does
+not correct for any host delays reported by the peer ({{ack-delay}}).
+
+An RTT sample MUST NOT be generated for a packet that is not newly acknowledged.
+This avoids multiple RTT samples for the same packet.
+
+An RTT sample MUST NOT be generated for a packet that is not the largest
+acknowledged in the received ACK frame.  A peer reports host delays for only the
+largest acknowledged packet in an ACK frame, which is assumed by subsequent
+computations of smoothed_rtt and rttvar in correcting for host delays.
+
+An RTT sample MUST NOT be generated if the ACK frame does not newly acknowledge
+an ack-eliciting packet. A peer does not send an ACK frame in response to
+receiving only non-ack-eliciting packets, and an ACK frame that is subsequently
+sent can include an arbitrarily large Ack Delay field.  Ignoring such ACK frames
+avoids complications in subsequent smoothed_rtt and rttvar computations.
+
+A sender might generate multiple RTT samples per RTT when multiple ACK frames
+are received within an RTT.  As suggested in {{?RFC6298}}, doing so might result
+in inadequate history in smoothed_rtt and rttvar.  This is an open question
+which we leave for future research to address.
+
+## Computing min_rtt {#min-rtt}
+
+min_rtt is the minimum RTT observed over the lifetime of the connection.
+min_rtt is set to the latest_rtt on the first sample in a connection, and to the
+lesser of min_rtt and latest_rtt on subsequent samples.
+
+An endpoint uses only locally observed times in computing the min_rtt and does
+not correct for host delays reported by the peer ({{ack-delay}}).  Doing so
+allows the endpoint to set a lower bound for the smoothed_rtt based entirely on
+what it observes (see {{smoothed-rtt}}), and limits potential underestimation
+due to erroneously-reported delays by the peer.
+
+## Computing smoothed_rtt and rttvar {#smoothed-rtt}
+
+smoothed_rtt is an exponentially-weighted moving average of an endpoint's
+RTT samples, and rttvar is the endpoint's estimated variance in
+the RTT samples.
+
+smoothed_rtt estimates path latency by correcting RTT samples for peer-reported
+host delays ({{ack-delay}}).  A peer indicates that the peer's delay in sending
```

Comment:
Check grammar.  "A peer indicates that the peer's delay" might be correct, but it's hard to read.